### ISSUE TITLE
swagger-ui fails to render using monolith application with service-discovery and jhipster-registry.

### ISSUE THREAD

**Issue by ongrieh** [jhipster/generator-jhipster#17531]
##### **Overview of the issue**

Swagger UI works fine with JHipster generated applications as long as no JHipster Registry is used. It seems that the `/management/jhiopenapigroups` provides an empty response. I am uncertain whether this is a JHipster Registry or a JHipster Generator issue. In the past JHipster Registry was always adjusted to updates of JHipster Generator.

##### **Motivation for or Use Case**

Being apple to us Swagger UI is an essential part of JHipster.

##### **Reproduce the error**

1. Create a JHipster application with a JHipster Registry
2. Start the JHipster application and make sure it connects to the JHipster Registry (Version 7.1)
3. Navigate to "API" under Administration in the application
![image](https://user-images.githubusercontent.com/45212418/148775707-ed484266-5b23-485a-914a-8c23f2b2041f.png)

Try to start your application without the JHipster Registry (fail-fast: false) and it works fine.

##### **JHipster Registry Version(s)**

7.2

##### **Browsers and Operating System**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "baseName": "openapitest",
    "jhipsterVersion": "7.5.0",
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "creationTimestamp": 1641810133975,
    "serviceDiscoveryType": "eureka",
    "reactive": false,
    "authenticationType": "jwt",
    "packageName": "com.mycompany.myapp",
    "serverPort": "8080",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": false,
    "nativeLanguage": "en",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "devServerPort": 4200,
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "enableGradleEnterprise": false,
    "languages": ["en", "fr"]
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)

git version 2.32.0.windows.2

node: v16.13.0

npm: 8.0.0

Docker version 20.10.11, build dea9396

Docker Compose version v2.2.1

No change to package.json was detected. No package manager install will be executed.

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by pascalgrimaud** [comment]
@ongrieh : not sure but I can be related to the migration from springfox to springdoc -> https://github.com/jhipster/generator-jhipster/pull/17194

**Comment by ongrieh** [comment]
Hi @pascalgrimaud, I agree to your assumption. The issue occurred after updating to version 7.5 and does not show for versions below. In addition time `/management/jhiopenapigroups` is newly introduced. 
Regards

**Comment by mshima** [comment]
From the screenshot it's frontend related.
What browser are you using?
We are using Promise.allSettled: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

**Comment by mshima** [comment]
There are 2 problems here.
1 - frontend should not fail to show the ui, in any case.
2 - jhipster-registry is failing to return the definition.

I could not reproduce your problem with default config, ex:
```
jhipster jdl --inline "application { config { baseName foo applicationType gateway authenticationType oauth2 }}"
npm docker:others:up
./mvnw
```
Looks like you have fine tuned timeouts.

In any case, for frontend:
I will create a PR improving errors.

For jhipster-registry:
v2 apis should be replace with v3 apis:
https://github.com/jhipster/jhipster-registry/blob/9065b7864abb92b87afc13797ce492976a51133d/src/main/java/tech/jhipster/registry/config/JWTSecurityConfiguration.java#L138
https://github.com/jhipster/jhipster-registry/blob/9065b7864abb92b87afc13797ce492976a51133d/src/main/java/tech/jhipster/registry/config/OAuth2SecurityConfiguration.java#L120

**Comment by ongrieh** [comment]
Hi @mshima, I can confirm that the given configuration

`jhipster jdl --inline "application { config { baseName foo applicationType gateway authenticationType oauth2 }}"`
`npm run docker:others:up`
`./mvnw`

works fine and as expected. But the setting I am using is more like:

` jhipster jdl --inline "application { config { baseName foo applicationType monolith authenticationType jwt serviceDiscoveryType eureka }}"`
`npm run docker:others:up`
`./mvnw`

This is also working fine as in dev mode. But than I took my given example a little further and tried to run it as production application. So...:
`./mvnw -ntp verify -DskipTests -Pprod jib:dockerBuild`
and
`docker:app:up`

I made sure the Foo application connected to the JHipster Registry 
![image](https://user-images.githubusercontent.com/45212418/148850238-e912084e-b50f-42e1-baeb-2cc0df0b0e11.png)
in prod mode
![image](https://user-images.githubusercontent.com/45212418/148850304-f6106b6e-e107-43d0-845c-87316dbfe9d7.png)
and I found the swagger ui of the application to fail again
![image](https://user-images.githubusercontent.com/45212418/148850455-ccb16ef9-c629-49a2-8612-12557ec2bb97.png)
(btw the swagger ui of the JHipster Registry works fine)

I did not try to run your example in prod but it occurs to me that it works when the application is in dev mode but fails for production. I am observing the same behavior for my actual use case.

Thanks again :)

**Comment by mshima** [comment]
Another issue should be opened to generate jhipster-registry.

**Comment by julillosamaral** [comment]
> Hi @pascalgrimaud, I agree to your assumption. The issue occurred after updating to version 7.5 and does not show for versions below. In addition time `/management/jhiopenapigroups` is newly introduced. Regards

I just posted this here: https://github.com/jhipster/generator-jhipster/pull/17533#issuecomment-1011412546

As I said in that comment, the endpoint `/services/${service}/management/jhiopenapigroups` in my service is returning me a 404 Not Found.

I moved from jhipster 7.0.1 to 7.5.0.

Why the registry is involved in this? Isn't the open api definition on the microservice that serves the api or is this something that changed?

Let me know if you need some more information from me. 

Thanks,

**Comment by julillosamaral** [comment]
I bit more of insight.

After posting the last message I started watching the PR created that was migrating away from springfox.
I noticed the jhipster command didn't create the file `com.katalyst.config.OpenApiConfiguration` so I did it manually.

After adding that class with the code taken from the PR adding springdoc the application started failing with a bean injection exception so regenerated again but accepting the changes in `application.yml`.

After that the application started, went to the UI and the documentation is working again.

Hope this helps!
